rio/src/RIO/Prelude/Extra.hs

```diff
@@ -40,6 +41,23 @@ mapMaybeM f = liftM catMaybes . mapM f
 forMaybeM :: Monad m => [a] -> (a -> m (Maybe b)) -> m [b]
 forMaybeM = flip mapMaybeM
 
+-- | Extend 'foldMap' to allow side effects.
```

Comment:
Uh, it seems this is going to be in `base` soon:

https://twitter.com/chessai2/status/1000859102954840065